### ISSUE TITLE
Loss is NaN after adding the callback

### ISSUE THREAD

**Issue by EliasVansteenkiste** [ar90n/pytorch-lightning-sam-callback#5]
I added the callback, trained with precision 16 and I get the following output during training.

<img width="673" alt="Screenshot 2022-08-14 at 22 48 50" src="https://user-images.githubusercontent.com/2494383/184554379-a6a4e78d-9e50-474a-aed6-cab287bc7cb5.png">

Training without the callback just gives values for the loss function.

**Comment by EliasVansteenkiste** [comment]
This ended up being an issue unrelated to the sam callback.

**Comment by bobyard-com** [comment]
Set gradient_clip_val=0.5, in trainer = Trainer(

**Comment by vinnik-dmitry07** [comment]
Set gradient_clip_val=0.5, in trainer = Trainer(
